### DISCUSSION TITLE
Using TypeSpec with JSON RPC?

### DISCUSSION THREAD

**Discussion by lukewestby** [microsoft/typespec#3257] [Q&A] [answered]
Hi folks,

I apologize if I've just missed something, but I'm struggling to find some information. The homepage says

> TypeSpec's standard library includes support for OpenAPI 3.0, JSON Schema 2020-12, Protobuf, and JSON RPC.

But I haven't been able to find packages or documentation related to JSON RPC anywhere. Where can I learn more about this support?

**Comment by bterlson** [comment, marked as answer]
Ooof, the homepage should be updated. JSON RPC is being worked on but unfortunately isn't ready at this time 😟 That reference comes from a time when we were optimistic it could be done in time.

**Reply by lukewestby** [reply]
Sounds good, thank you! Looking forward to that support, thanks for all your hard work.

**Comment by sonatard** [comment]
@bterlson Is there any progress on the plan? Currently, MCP is gaining popularity due to AI Agents, and it utilizes JSON-RPC v2.
I would be happy if TypeSpec could be used to make the specification type-safe when implementing MCP myself.

**Reply by bterlson** [reply]
JSON RPC has been put on the back burner but is still something we want to tackle. That said, MCP codegen is definitely something we're interested in and we're prototyping some stuff as we speak. Rather than code generating against the raw protocol, it sits on top of the low-level TS SDK. Presumably a similar thing could be done with Python. It still provides a type safe boundary with nice, clean (non-Zod-inferred) interfaces. If you're curious you can follow along [here](https://github.com/bterlson/typespec-mcp), but note that everything is super proof of concept-y at this time.
